server/src/main/java/org/opensearch/cluster/routing/RoutingPool.java

```diff
@@ -58,6 +60,11 @@ public static RoutingPool getShardPool(ShardRouting shard, RoutingAllocation all
      * @return {@link RoutingPool} for the given index.
      */
     public static RoutingPool getIndexPool(IndexMetadata indexMetadata) {
-        return indexMetadata.isRemoteSnapshot() ? REMOTE_CAPABLE : LOCAL_ONLY;
+        return indexMetadata.isRemoteSnapshot()
+            || (FeatureFlags.isEnabled(FeatureFlags.TIERED_REMOTE_INDEX)
+                && IndexModule.DataLocalityType.PARTIAL.name()
+                    .equals(indexMetadata.getSettings().get(IndexModule.INDEX_STORE_LOCALITY_SETTING.getKey())))
+                        ? REMOTE_CAPABLE
+                        : LOCAL_ONLY;
```

Comment:
nit: This is a little confusing to look at, I think the `if (FeatureFlags.enabled) {...}` semantic is generally cleaner

Reply:
will refactor this a bit